packages/@dynamic-labs-connectors/sequence-cross-app-evm/src/SequenceCrossAppConnector.ts

```diff
@@ -0,0 +1,301 @@
+import { type EthereumWalletConnectorOpts } from "@dynamic-labs/ethereum-core";
+import {
+  EthereumInjectedConnector,
+  type IEthereum,
+} from "@dynamic-labs/ethereum";
+import { DynamicError } from "@dynamic-labs/utils";
+import {
+  WalletMetadata,
+  type Chain,
+} from "@dynamic-labs/wallet-connector-core";
+import { findWalletBookWallet } from "@dynamic-labs/wallet-book";
+import { toHex, getAddress, TransactionRejectedRpcError } from "viem";
+import { ethers } from "ethers";
+import { allNetworks, EIP1193Provider } from "@0xsequence/network";
+import { WalletConnector } from "@dynamic-labs/wallet-connector-core";
+
+import { ProviderTransport } from "./providerTransport";
+
+export const createSequenceCrossAppConnector = (
+  metadata: WalletMetadata,
+  transportConfig: CrossAppTransportConfig
+): [new (props: EthereumWalletConnectorOpts & { transportConfig: CrossAppTransportConfig }) => WalletConnector] => {
+  return [
+    class extends SequenceCrossAppConnector {
+      constructor(
+        props: EthereumWalletConnectorOpts & {
+          transportConfig: CrossAppTransportConfig;
+        }
+      ) {
+        super({
+          ...props,
+          metadata,
+          transportConfig,
+        });
+      }
+    },
+  ];
+};
+
+export type CrossAppTransportConfig = {
+  projectAccessKey: string;
+  walletUrl: string;
+  initialChainId: number;
+};
+
+export class SequenceCrossAppConnector extends EthereumInjectedConnector {
+  private walletName: string;
+
+  private nodesUrl = "https://nodes.sequence.app";
+
+  sequenceWaasTransportProvider: SequenceWaasTransportProvider;
+
+  /**
+   * The name of the wallet connector
+   * @override Required override from the base connector class
+   */
+  override get name() {
+    return this.walletName;
+  }
+
+  /**
+   * The constructor for the connector, with the relevant metadata
+   * @param props The options for the connector
+   */
+  constructor(
+    props: EthereumWalletConnectorOpts & {
+      transportConfig: CrossAppTransportConfig;
+    }
+  ) {
+    super({
+      ...props,
+    });
+
+    this.walletName = props.metadata!.id;
+
+    this.sequenceWaasTransportProvider = new SequenceWaasTransportProvider(
+      props.transportConfig.projectAccessKey,
+      props.transportConfig.walletUrl,
+      props.transportConfig.initialChainId,
+      this.nodesUrl
+    );
+
+    this.wallet = findWalletBookWallet(this.walletBook, this.key);
+  }
+
+  override supportsNetworkSwitching(): boolean {
+    return false;
+  }
+
+  override isInstalledOnBrowser(): boolean {
+    return true;
+  }
+
+  override async init(): Promise<void> {
+    this.walletConnectorEventsEmitter.emit("providerReady", {
+      connector: this,
+    });
+  }
```

Comment:
I don't think you need to emit the event here, since you're not waiting for the provider to be initialized. You can probably just remove this override